docs/v3/how-to-guides/self-hosted/docker-compose.mdx

```diff
@@ -0,0 +1,145 @@
+---
+title: Run Prefect Server with Docker Compose
+sidebarTitle: Prefect Server with Docker Compose
+description: Deploy a self-hosted Prefect server environment using Docker Compose, including PostgreSQL, Redis, server, services, and worker.
+---
+
+This guide shows how to run a complete self-hosted Prefect server environment using Docker Compose. This setup includes the following services:
+
+- PostgreSQL database  
+- Redis message broker  
+- Prefect server  
+- Prefect background services  
+- Prefect worker  
+
+## Docker Compose file
+
+Create a `compose.yml` file with the following contents:
+
+```yaml
+services:
+  postgres:
+    image: postgres:14
+    environment:
+      POSTGRES_USER: prefect
+      POSTGRES_PASSWORD: prefect
+      POSTGRES_DB: prefect
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U prefect"]
+      interval: 5s
+      timeout: 5s
+      retries: 5
+
+  redis:
+    image: redis:7
+    volumes:
+      - redis_data:/data
+
+  prefect-migrate:
+    image: prefecthq/prefect:3-latest
+    environment:
+      PREFECT_API_DATABASE_CONNECTION_URL: postgresql+asyncpg://prefect:prefect@postgres:5432/prefect
+    depends_on:
+      postgres:
+        condition: service_healthy
+    command: >
+      bash -c "prefect server database upgrade || true"
+
+  prefect-server:
+    image: prefecthq/prefect:3-latest
+    depends_on:
+      postgres:
+        condition: service_healthy
+      prefect-migrate:
+        condition: service_completed_successfully
+    environment:
+      PREFECT_API_DATABASE_CONNECTION_URL: postgresql+asyncpg://prefect:prefect@postgres:5432/prefect
+      PREFECT_SERVER_API_HOST: 0.0.0.0
+      PREFECT_API_URL: http://localhost:4200/api
+      PREFECT_MESSAGING_BROKER: prefect_redis.messaging
+      PREFECT_MESSAGING_CACHE: prefect_redis.messaging
+      PREFECT_REDIS_MESSAGING_HOST: redis
+      PREFECT_REDIS_MESSAGING_PORT: 6379
+      PREFECT_REDIS_MESSAGING_DB: 0
+      MIGRATE_ON_START: "false"
```

Comment:
this would be `PREFECT_SERVER_MIGRATE_ON_START` but on second thought, i think we can omit this for now - sorry for the whiplash
```suggestion
```